synapse/handlers/device.py

```diff
@@ -309,6 +309,15 @@ async def on_federation_query_user_devices(self, user_id: str) -> JsonDict:
             "self_signing_key": self_signing_key,
         }
 
+    async def handle_room_un_partial_stated(self, room_id: str) -> None:
+        """Handles sending appropriate device list updates in a room that has
+        gone from partial to full state.
+        """
+
+        logger.error(
```

Comment:
Is there anything we need to fix for worker mode or is this simply unreachable?
```suggestion

        # TODO(faster_joins): worker mode support
        #   https://github.com/matrix-org/synapse/issues/12994
        logger.error(
```

Reply:
> Is there anything we need to fix for worker mode or is this simply unreachable?

This will be hit if you try and do a partial join in worker mode; it never otherwise gets hit.

To make it work correctly i think we need to add a replication api to hit master (as currently only master can write device list updates), or change the device list stream to allow multi-worker support.